torch/compiler/config.py

```diff
@@ -0,0 +1,114 @@
+"""
+This is the top-level configuration module for the compiler, containing
+cross-cutting configuration options that affect all parts of the compiler
+stack.
+
+You may also be interested in the per-component configuration modules, which
+contain configuration options that affect only a specific part of the compiler:
+
+* :mod:`torch._dynamo.config`
+* :mod:`torch._inductor.config`
+* :mod:`torch._functorch.config`
+* :mod:`torch.fx.experimental.config`
+"""
+
+import sys
+from typing import Optional
+
+# NB: Docblocks go UNDER variable definitions!  Use spacing to make the
+# grouping clear.
+
+cache: bool = True
+"""When ``True`` (default), enable all (non-experimental) caches supported by
+the compiler, persisting compilation products across multiple invocations of
+the compiler.  When ``False``, do not enable caches by default (however, if an
+individual cache is explicitly enabled, we will respect those settings.)  By
+default, caches are written local filesystem at
+``/tmp/torchinductor_$USERNAME``.
+
+At time of writing (PyTorch 2.5), the following caches are enabled by default:
+
+* Triton cache, which caches cubins generated by the Triton compiler
+  (technically, this cache is implemented by the Triton compiler and not by the
+  PyTorch compiler, but this configuration option does control whether or not
+  the Triton cache is used.)
+
+* Inductor autotune cache, which caches autotuning results from Inductor
+  compilation (this includes tuning for both Inductor generated kernels as
+  well as matrix multiply algorithm selection, among others).  This cache is
+  controlled by :data:`torch._inductor.config.autotune_local_cache`,
+  :data:`torch._inductor.config.autotune_remote_cache`
+  (:envvar:`TORCHINDUCTOR_AUTOTUNE_REMOTE_CACHE`) and
+  :data`torch._inductor.config.bundled_autotune_remote_cache`
+  (:envvar:`TORCHINDUCTOR_BUNDLED_AUTOTUNE_REMOTE_CACHE`).
```

Comment:
I guess here and below, how do these different options interact with each other?